### ISSUE TITLE
detekt baseline not catching all existing issues

### ISSUE THREAD

**Issue by ryanloftus02** [detekt/detekt#5559]
## Expected Behavior

when I run `./gradlew detektInternalDebug detektDebug` immediately after running `./gradlew detektBaselineInternalDebug detektBaselineDebug`, detekt should succeed

## Observed Behavior

when I run `./gradlew detektInternalDebug detektDebug` immediately after running `./gradlew detektBaselineInternalDebug detektBaselineDebug`, detekt fails

## Steps to Reproduce

1. Run `./gradlew detektBaselineInternalDebug detektBaselineDebug`
2. Run `./gradlew detektInternalDebug detektDebug`

## Context

I'm trying to restrict a certain function from being called in my project. This required changing from using `detekt` to using `detekt<variant>` to enable type resolution. However, when I generate the baselines, it doesn't seem to catch all the existing issues because running the new detekt command fails.

The project has several modules, each has the following in its `build.gradle` file:
```
detekt {
    // ...
    baseline = file("$rootDir/MainModule/detekt-baseline.xml")
    // ...
}
```

## Your Environment

* Version of detekt used: 1.19.0
* Version of Gradle used (if applicable): 7.4.2
* Version of Kotlin used: 1.7.20
* Operating System and version: macOS Monterey 12.6.1

**Comment by 3flex** [comment]
Can you please try again with the latest detekt release?

A reproducer project will also help, or providing a link to your project if open source.

**Comment by ryanloftus02** [comment]
> Can you please try again with the latest detekt release?

Just tried again with detekt version 1.22.0 and the issue still occurs. Unfortunately I can't provide a link to the project, but based on the output from the failed detekt command it seems like its failing from rules that don't use type resolution.

**Comment by cortinico** [comment]
> As a side note, is the command I've been using the correct way to run detekt with type resolution / generate baselines for all modules with either an `InternalDebug` or `Debug` variant?

Yes that's the correct one.
If you could create a small repro with the debug/internaldebug flavor, it would be really beneficial

**Comment by ryanloftus02** [comment]
Okay, I think I've solved the issue. As I mentioned in my description, I am using one baseline file for all modules in the project. It seems that the modules were rewriting the baseline instead of adding to it when I ran the detekt command at the root of the project. This meant that only one module's baseline issues were reported in the baseline file. I was able to get this to work by changing the `build.gradle` in each module to give each module its own baseline file.
